packages/tracing/src/index.ts

```diff
@@ -0,0 +1,92 @@
+import { context, propagation, SpanStatusCode, trace } from '@opentelemetry/api';
+import type { Span, SpanOptions, Tracer } from '@opentelemetry/api';
+import { OTLPTraceExporter } from '@opentelemetry/exporter-trace-otlp-grpc';
+import { NodeSDK } from '@opentelemetry/sdk-node';
+
+export { initDatabaseTracing } from './traceDatabaseCalls';
+
+let tracer: Tracer;
+
+export function isTracingEnabled() {
+	return ['yes', 'true'].includes(String(process.env.TRACING_ENABLED).toLowerCase());
+}
+
+export const startTracing = ({ service }: { service: string }) => {
+	const exporter = new OTLPTraceExporter();
+
+	const sdk = new NodeSDK({
+		traceExporter: exporter,
+		instrumentations: [],
+		serviceName: service,
+	});
+	sdk.start();
+
+	tracer = trace.getTracer(service);
+};
+
+export function tracerSpan<F extends (span?: Span) => ReturnType<F>>(
+	name: string,
+	options: SpanOptions,
+	fn: F,
+	optl?: unknown,
+): ReturnType<F> {
+	if (!isTracingEnabled()) {
+		return fn();
+	}
+
+	const computeResult = (span: Span) => {
+		try {
+			const result = fn(span);
+			if (result instanceof Promise) {
+				result.catch((err) => {
+					span.recordException(err);
+					span.setStatus({
+						code: SpanStatusCode.ERROR,
+						message: err.message,
+					});
+				});
+
+				return result;
+			}
+			return result;
+		} catch (err: any) {
+			span.recordException(err);
+			span.setStatus({
+				code: SpanStatusCode.ERROR,
+				message: err.message,
+			});
+			throw err;
+		} finally {
+			span.end();
+		}
+	};
+
+	if (optl) {
+		const activeContext = propagation.extract(context.active(), optl);
+
+		return tracer.startActiveSpan(name, options, activeContext, computeResult);
```

Comment:
We must make sure this is defined, right? What if someone forgets to start it?

Reply:
yeah, we can test if it is defined and provide a better error than just node breaking because `tracer` is `undefined`.. I'll do that